### DISCUSSION TITLE
async2 - The .NET Runtime Async experiment concludes

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#117] [General]
# async2 - The .NET Runtime Async experiment concludes

The .NET team has been working on a new experiment called async2, which is a new implementation of the async/await pattern that is designed to be more efficient and more flexible than the current implementation. It started with green threads and ended with an experiment that moves async and await to the runtime. This post will cover the journey of async2 and the conclusion of the experiment.

https://steven-giesel.com/blogPost/59752c38-9c99-4641-9853-9cfa97bb2d29/async2-the-net-runtime-async-experiment-concludes

**Comment by toupswork** [comment]
Thank God .NET has always used OS threads.
